source/adios2/core/Engine.cpp

```diff
@@ -37,7 +37,8 @@ Mode Engine::OpenMode() const noexcept { return m_OpenMode; }
 
```

Comment:
My view is that it serves no purpose in BeginStep().  A mode is there in the API because we share that with the writer and the writer has modes in BeginStep() (or at least some writers do), however the reader just has one.  I've left it named NextAvailable, but maybe we should change the name?  Just Next?  Or simply Read?

Reply:
I think either way makes better sense than leaving it NextAvailable, because in many cases it's obviously not reading the next available step.

Reply:
OK, I'd vote for Mode::Read.  Happy to hear other opinions...

Reply:
Read makes more sense to me because 1) I am unable to imagine a scenario where a user believes that selecting that mode implies any more specific behavior for a Reader, and 2) there is no change of a user believing that might be a valid mode on a Writer.

What would you think of adding a dummy StepMode::Read instead of reusing Mode::Read?